cmd/maya-apiserver/app/server/snapshot_endpoint_v1alpha1.go

```diff
@@ -0,0 +1,171 @@
+		return volOp.snapshotList(volName)
+	default:
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+}
+
+// SnapshotCreate is http handler which handles snaphsot-create request
+func (v *volumeAPIOpsV1alpha1) snapshotCreate() (interface{}, error) {
+
+	if v.req.Method != "PUT" && v.req.Method != "POST" {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	glog.Infof("Processing Volume create snapshot request")
+
+	snap := v1.VolumeSnapshot{}
+
+	// The yaml/json spec is decoded to VolumeSnapshot struct
+	if err := decodeBody(v.req, &snap); err != nil {
+
+		return nil, CodedError(400, err.Error())
+	}
+
+	// SnapshotName is expected to be available even in the minimalist specs
+	if snap.Metadata.Name == "" {
+		return nil, CodedError(400, fmt.Sprintf("Error: snapshot name missing in '%v'", snap.SnapshotName))
+	}
+
+	// Name is expected to be available even in the minimalist specs
+	if snap.Spec.VolumeName == "" {
+
+		return nil, CodedError(400, fmt.Sprintf("Error: volume name missing in '%v'", snap.Spec.VolumeName))
+	}
+
+	glog.Infof("Processing snapshot-create request of volume: %s", snap.Spec.VolumeName)
+
+	voldetails, err := v.read(snap.Spec.VolumeName)
+	if err != nil {
+		return nil, err
+	}
+
+	ControllerIP := voldetails.Annotations["openebs.io/controller-ips"]
+
+	var labelMap map[string]string
+	snapinfo, err := jiva.Snapshot(snap.Metadata.Name, ControllerIP, labelMap)
+	if err != nil {
+		glog.Errorf("Failed to create snapshot of volume %v : %v", snap.Spec.VolumeName, err)
+		return nil, err
+	}
+
+	glog.Infof("Snapshot created for volume [%s] is [%s]\n", snap.Spec.VolumeName, snap.Metadata.Name)
+
+	return snapinfo, nil
+}
+
+// SnapshotRevert is http handler for handling snapshot-revert request.
+// Volume and existing snapshot name will be passed as struct fields to
+// revert to that particular snapshot
+func (v *volumeAPIOpsV1alpha1) snapshotRevert() (interface{}, error) {
+	if v.req.Method != "PUT" && v.req.Method != "POST" {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+	glog.Infof("Processing Volume snapshot-revert request")
+
+	snap := v1.VolumeSnapshot{}
+
+	// The yaml/json spec is decoded to VolumeSnapshot struct
+	if err := decodeBody(v.req, &snap); err != nil {
+
+		return nil, CodedError(400, err.Error())
+	}
+
+	// SnapshotName is expected to be available even in the minimalist specs
+	if snap.Metadata.Name == "" {
+		return nil, CodedError(400, fmt.Sprintf("ERROR: Snapshot name missing in '%v'", snap.Metadata.Name))
+	}
+
+	// Name is expected to be available even in the minimalist specs
+	if snap.Spec.VolumeName == "" {
+
+		return nil, CodedError(400, fmt.Sprintf("ERROR: Volume name missing in '%v'", snap))
+	}
+
+	glog.Infof("Processing snapshot-revert request of volume: %s", snap.Spec.VolumeName)
+
+	voldetails, err := v.read(snap.Spec.VolumeName)
+	if err != nil {
+		return nil, err
+	}
+
+	ControllerIP := voldetails.Annotations["openebs.io/controller-ips"]
```

Comment:
can this refer to targetIP property of the volume rather than making use of annotation?
Can jiva API calls be via service IP?